docs/resources/integration.md

```diff
@@ -0,0 +1,103 @@
+---
+layout: "fastly"
+page_title: "Fastly: integration"
+sidebar_current: "docs-fastly-resource-integration"
+description: |-
+  Provides a Fastly integration. Fastly will use these integrations to send you notifications and alerts related to the Fastly product. Mailing list, Microsoft Teams, New Relic, PagerDuty, Slack, and webhook are supported as integrations.
+---
+
+# fastly_integration
+
+Provides a Fastly integration. Fastly will use these integrations to send you notifications and alerts related to the Fastly product. Mailing list, Microsoft Teams, New Relic, PagerDuty, Slack, and webhook are supported as integrations.
+
+## Example Usage
+
+```terraform
+# IMPORTANT: mailing list integrations require confirmation.
+# To send a confirmation email and verify integration status,
+# after applying changes using Terraform, please visit
+# https://manage.fastly.com/observability/alerts/integrations
```

Comment:
Good question. It would likely be confusing for users who don't know about the extra step. To improve that experience, I added a diagnostic warning that will appear after applying or refreshing when there's an unconfirmed mailing list integration.

Reply:
The Fastly API does not eventually return an API error; that resource just exists with one of several unconfirmed statuses `needs_confirmation`, `confirmation_pending`, or `confirmation_expired`. A further potential enhancement would be for Fastly Terraform provider to request a confirmation email automatically and save the user from having to visit a UI, which would simplify the workflow if the Terraform user doesn't have a Fastly login. There's a little complexity with where to put such a hook though to deduplicate the requests per email address for the case when there are multiple mailing list integrations with the same address.